Files/View Models/ItemViewModel.cs

```diff
@@ -306,7 +306,7 @@ public ItemViewModel()
             FilesAndFolders = new ReadOnlyObservableCollection<ListedItem>(_filesAndFolders);
             _addFilesCTS = new CancellationTokenSource();
             _semaphoreCTS = new CancellationTokenSource();
-
+            shouldDisplayFileExtensions = App.AppSettings.ShowFileExtensions;
```

Comment:
Why you changed one bool to another one?

Reply:
Because it is apparently redundant to check the value of the property for each file being added. What this does is check the value of that property from AppSettings only when it would need to be updated.

Reply:
A lot of CPU time was being spent for each file by resolving the Get() method for App.AppSettings.ShowFileExtensions. This stores that property's value and updates it once per enumeration rather than once per file.